lib/uring/heap.ml

```diff
@@ -68,7 +68,52 @@ let create : type a. int -> a t =
 
 exception No_space
 
-let alloc t data ~extra_data =
+let in_use t = t.in_use
+
+let release t =
+  if t.in_use > 0 then invalid_arg "Heap still in use!"
+  else if t.in_use < 0 then invalid_arg "Heap already released!";
+  t.in_use <- -100;
+  t.free_head <- free_list_nil
+
+let is_released t = t.in_use < 0
+
+let grow t =
```

Comment:
I guess there's a precondition here that the array is currently full? Might be worth stating it explicitly.